### PR TITLE
 [MO] - [system test] -> zookeeper upgrade test with additional check

### PR DESCRIPTION
**Pull request by Seequick1**
### Type of change

- Enhancement / new feature
- Refactoring

### Description

This PR moves Zookeeper upgrade tests to one specific ST called ZookeeperUpgradeST. Also adding changing `log.message.format.version` to newer version. Morever fixing downgrade test and upgrade problem with timeout.
### Checklist

- [x] Write tests
- [ ] Make sure all tests pass

### CODE REVIEW

**Comment by Seequick1** [comment]
@strimzi-ci run tests profile=upgrade testcase=ZookeeperUpgradeST test_only

**Comment by Seequick1** [comment]
@strimzi-ci run tests profile=upgrade testcase=ZookeeperUpgradeST test_only

**Comment by scholzj** [comment]
@Seequick1 Could you please rabese this to resolve the conflicts?

File: systemtest/src/main/java/io/strimzi/systemtest/utils/kubeUtils/objects/PodUtils.java [hunk 1]
```diff
@@ -166,7 +166,15 @@ public static void waitUntilPodContainersCount(String podNamePrefix, int numberO
         TestUtils.waitFor("Waiting till pod" + podNamePrefix + " will have " + numberOfContainers + " containers",
             Constants.GLOBAL_POLL_INTERVAL, Constants.GLOBAL_STATUS_TIMEOUT,
             () -> kubeClient().listPodsByPrefixInName(podNamePrefix).get(0).getSpec().getContainers().size() == numberOfContainers);
-        LOGGER.info("Waiting till pod {} will have {} containers", podNamePrefix, numberOfContainers);
+        LOGGER.info("Pod {} has {} containers", podNamePrefix, numberOfContainers);
+    }
+
+    public static void waitUntilPodReplicasCount(String podNamePrefix, int exceptedPods) {
```
Comment (Frawless) [inline systemtest/src/main/java/io/strimzi/systemtest/utils/kubeUtils/objects/PodUtils.java]: It's not `PodReplicasCount` it's just `ContainerInPodCount`, better to use `waitForPodsReady()` I think or at least name it correctly.

File: systemtest/src/test/java/io/strimzi/systemtest/upgrade/ZookeeperUpgradeST.java [hunk 2]
```diff
@@ -0,0 +1,187 @@
...
+
+        // Wait for the zk version change roll
+        zkPods = StatefulSetUtils.waitTillSsHasRolled(KafkaResources.zookeeperStatefulSetName(CLUSTER_NAME), zkReplicas, zkPods);
+        LOGGER.info("1st Zookeeper roll (image change) is complete");
+
+        // Wait for the kafka broker version change roll
+        kafkaPods = StatefulSetUtils.waitTillSsHasRolled(KafkaResources.kafkaStatefulSetName(CLUSTER_NAME), kafkaReplicas, kafkaPods);
+        LOGGER.info("Kafka roll (image change) is complete");
+
+        PodUtils.waitUntilPodReplicasCount(KafkaResources.kafkaStatefulSetName(CLUSTER_NAME), 2);
```
Comment (scholzj) [inline systemtest/src/test/java/io/strimzi/systemtest/upgrade/ZookeeperUpgradeST.java]: You run with 3 replicas, so you should wait for 3 pods (the same below as well).

File: systemtest/src/test/java/io/strimzi/systemtest/upgrade/ZookeeperUpgradeST.java [hunk 3]
```diff
@@ -0,0 +1,187 @@
...
+
+        // Wait for the zk rolling update
+        StatefulSetUtils.waitTillSsHasRolled(KafkaResources.kafkaStatefulSetName(CLUSTER_NAME), kafkaReplicas, kafkaPods);
+        LOGGER.info("2nd Kafka roll (update) is complete");
+
+        PodUtils.waitUntilPodReplicasCount(KafkaResources.kafkaStatefulSetName(CLUSTER_NAME), 2);
+
+        StatefulSetUtils.waitTillSsHasRolled(KafkaResources.kafkaStatefulSetName(CLUSTER_NAME), kafkaReplicas, zkPods);
```
Comment (scholzj) [inline systemtest/src/test/java/io/strimzi/systemtest/upgrade/ZookeeperUpgradeST.java]: This is a bit confusing. The message bellow says _Zookeeper roll_. But this method seems to check Kafka.

Comment (tomncooper) [reply systemtest/src/test/java/io/strimzi/systemtest/upgrade/ZookeeperUpgradeST.java]: This does seem confusing. When I designed the original ST the sequence was ZK Roll -> Kafka Roll -> ZK Roll then the update was complete. This seems to be ZK Roll -> Kafka Roll -> ZK Roll -> Kafka Roll -> Kafka Roll? Was the a change introduced by some new Kafka Rolling code? Or are there setup changes that require more rolls?

Comment (scholzj) [reply systemtest/src/test/java/io/strimzi/systemtest/upgrade/ZookeeperUpgradeST.java]: I think that between two minor versions there were always 2 Kafka rolling updates. The third one is related to the configuration from file and how is Fabric 8 / Kubernetes seemingly randomly changing the different HereDoc styles in tha patched YAML.

Comment (scholzj) [reply systemtest/src/test/java/io/strimzi/systemtest/upgrade/ZookeeperUpgradeST.java]: (The third should be solved with the dynamic reconfiguration)

**Comment by Seequick1** [comment]
@strimzi-ci run tests profile=upgrade testcase=ZookeeperUpgradeST test_only

File: systemtest/src/test/java/io/strimzi/systemtest/upgrade/ZookeeperUpgradeST.java (hunk 3 above)
Comment (Seequick1) [reply systemtest/src/test/java/io/strimzi/systemtest/upgrade/ZookeeperUpgradeST.java]: I have tested it many times. 
```
1) Upgrade has following approach - Z roll, K roll, Z roll
2) Downgrade has - Z roll, K roll, K roll, K roll, K roll
```

Everything passing. Executed few times.

File: systemtest/src/test/java/io/strimzi/systemtest/upgrade/ZookeeperUpgradeST.java [hunk 4]
```diff
@@ -0,0 +1,188 @@
...
+
+        // Wait for the zk version change roll
+        zkPods = StatefulSetUtils.waitTillSsHasRolled(KafkaResources.zookeeperStatefulSetName(CLUSTER_NAME), zkReplicas, zkPods);
+        LOGGER.info("1st Zookeeper roll (image change) is complete");
+
+        // Wait for the kafka broker version change roll
+        kafkaPods = StatefulSetUtils.waitTillSsHasRolled(KafkaResources.kafkaStatefulSetName(CLUSTER_NAME), kafkaReplicas, kafkaPods);
+        LOGGER.info("Kafka roll (image change) is complete");
+
+        if (testInfo.getDisplayName().contains("Downgrade")) {
```
Comment (tomncooper) [inline systemtest/src/test/java/io/strimzi/systemtest/upgrade/ZookeeperUpgradeST.java]: You could also just compare the intial and new Kafka versions: `if (initialVersion.compareTo(newVersion) > 0)`

Comment (Seequick1) [reply systemtest/src/test/java/io/strimzi/systemtest/upgrade/ZookeeperUpgradeST.java]: Thanks. This `Downgrade` and `Upgrade` is maybe little bit more straight looking.

**Pull request merged**